src/database.py

```diff
@@ -149,3 +149,160 @@ def write_fraud_event_to_database(event, db_connection):
                 'Failed to store a fraud event [Event ID {0}] due to integrity error'.format(event.event_id))
             raise integrityError
+
+
+def write_import_session(upload_session, db_connection, logger):
+    try:
+        with RunInTransaction(db_connection) as cursor:
+            cursor.execute("""
+                INSERT INTO idp_data.upload_sessions
+                (
+                    time_stamp,
+                    source_file_name,
+                    idp_entity_id,
+                    userid,
+                    passed_validation
+                )
+                VALUES
+                (
+                    %s,
+                    %s,
+                    %s,
+                    %s,
+                    FALSE
+                )
+                RETURNING id
+            """, [upload_session.time_stamp,
+                  upload_session.source_file_name,
+                  upload_session.idp_entity_id,
+                  upload_session.userid])
+
+            result = cursor.fetchone()
+            upload_session.id = result[0]
+            return upload_session
+
+    except KeyError as keyError:
+        logger.exception(
+            'Failed to store upload session for file {}, idp = {} due to key error'.format(
+                upload_session.source_file_name, upload_session.idp_entity_id)
+        )
+        raise keyError
+    except IntegrityError as integrityError:
+        logger.exception(
+            'Failed to store upload session for file {}, idp = {} due to integrity error'.format(
+                upload_session.source_file_name, upload_session.idp_entity_id)
+        )
+        raise integrityError
+
+
+def write_idp_fraud_event_to_database(upload_session, idp_fraud_event, cursor, logger):
+    try:
+        cursor.execute("""
+             INSERT INTO idp_data.idp_fraud_events
+             (
+                idp_entity_id,
+                idp_event_id,
+                time_stamp,
+                fid_code,
+                request_id,
+                pid,
+                client_ip_address,
+                contra_score,
+                event_id,
+                upload_session_id
+             )
+             VALUES
+             (
+                %(idp_entity_id)s,
+                %(idp_event_id)s,
+                %(timestamp)s,
+                %(fid_code)s,
+                %(request_id)s,
+                %(pid)s,
+                %(client_ip_address)s,
+                %(contra_score)s,
+                (
+                    SELECT event_id
+                      FROM billing.fraud_events
+                     WHERE fraud_event_id = %(idp_event_id)s
+                       AND entity_id = %(idp_entity_id)s
+                     LIMIT 1
+                ),
+                %(session_id)s
+             )
+             ON CONFLICT (idp_entity_id, idp_event_id)
+             DO UPDATE SET contra_score = EXCLUDED.contra_score + idp_fraud_events.contra_score
```

Comment:
Decision: for now, we'll remove the relevant index, so addition of contra scores will go away. We've decided we can't assume what the IDP means, and we may need to revisit should we get data from Post Office / SecureIdentity anyway.